src/server/field.rs

```diff
@@ -138,63 +125,42 @@ pub struct ContentDisp {
 }
 
 impl ContentDisp {
-    fn parse(headers: &[StrHeader]) -> Option<ContentDisp> {
+    fn parse(headers: &[StrHeader]) -> Result<ContentDisp, ParseHeaderError> {
         if headers.is_empty() {
-            return None;
+            return Err(ParseHeaderError::Missing);
         }
 
         const CONT_DISP: &'static str = "Content-Disposition";
-
-        let header = try_opt!(
-            find_header(headers, CONT_DISP),
-            error!("Field headers did not contain Content-Disposition header (required)")
-        );
+        let header = try!(find_header(headers, CONT_DISP).ok_or(ParseHeaderError::Missing));
 
         const NAME: &'static str = "name=";
         const FILENAME: &'static str = "filename=";
 
         let after_disp_type = {
-            let (disp_type, after_disp_type) = try_opt!(
-                split_once(header.val, ';'),
-                error!("Expected additional data after Content-Disposition type, got {:?}",
-                header.val)
-            );
-
-
+            let (disp_type, after_disp_type) = try!(split_once(header.val, ';').ok_or(ParseHeaderError::Invalid));
             if disp_type.trim() != "form-data" {
-                error!("Unexpected Content-Disposition value: {:?}", disp_type);
-                return None;
-            };
-
+                return Err(ParseHeaderError::Invalid);
```

Comment:
I'd like to have this contain the error message that was previously being logged, otherwise there's not much context for the error.